pkg/controller/nodeconfig/resource.go

```diff
@@ -116,6 +116,165 @@ func makeNodeConfigClusterRoleBinding() *rbacv1.ClusterRoleBinding {
 	}
 }
 
+func makeNodeDiskSetupDaemonSet(nc *scyllav1alpha1.NodeConfig, operatorImage string) *appsv1.DaemonSet {
+	if nc.Spec.LocalDiskSetup == nil {
+		return nil
+	}
+
+	labels := map[string]string{
+		"app.kubernetes.io/name":   naming.NodeConfigAppName,
+		naming.NodeConfigNameLabel: nc.Name,
+	}
+
+	return &appsv1.DaemonSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-node-setup", nc.Name),
+			Namespace: naming.ScyllaOperatorNodeTuningNamespace,
+			Labels:    labels,
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(nc, nodeConfigControllerGVK),
+			},
+		},
+		Spec: appsv1.DaemonSetSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: labels,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: labels,
+				},
+				Spec: corev1.PodSpec{
+					ServiceAccountName: naming.NodeConfigAppName,
+					NodeSelector:       nc.Spec.Placement.NodeSelector,
+					Affinity:           &nc.Spec.Placement.Affinity,
+					Tolerations:        nc.Spec.Placement.Tolerations,
+					Volumes: []corev1.Volume{
+						{
+							Name: "hostfs",
+							VolumeSource: corev1.VolumeSource{
+								HostPath: &corev1.HostPathVolumeSource{
+									Path: "/",
+									Type: pointer.Ptr(corev1.HostPathDirectory),
+								},
+							},
+						},
+					},
+					Containers: []corev1.Container{
+						{
+							Name:            naming.NodeConfigAppName,
+							Image:           operatorImage,
+							ImagePullPolicy: corev1.PullIfNotPresent,
+							Command: []string{
+								"/usr/bin/bash",
+								"-euExo",
+								"pipefail",
+								"-c",
+							},
+							Args: []string{
+								`
+cd "$( mktemp -d )"
+
+for f in $( find /host -mindepth 1 -maxdepth 1 -type d -printf '%f\n' ); do
+	mkdir -p "./${f}"
+	mount --rbind "/host/${f}" "./${f}"
+done
+
+for f in $( find /host -mindepth 1 -maxdepth 1 -type f -printf '%f\n' ); do
+	touch "./${f}"
+	mount --bind "/host/${f}" "./${f}"
+done
+
+find /host -mindepth 1 -maxdepth 1 -type l -exec cp -P "{}" ./ \;
+
+mkdir -p ./scylla-operator
+touch ./scylla-operator/scylla-operator
+mount --bind /usr/bin/scylla-operator ./scylla-operator/scylla-operator
+
+for f in ca.crt token; do
+	touch "./scylla-operator/${f}"
+	mount --bind "/var/run/secrets/kubernetes.io/serviceaccount/${f}" "./scylla-operator/${f}"
+done
+
+cat <<EOF > ./scylla-operator/kubeconfig
```

Comment:
why building your own kubeconfig instead of mounting the one that's already in /var/run ?

Reply:
it may not technically be there but if you mount the files where the need to be kube clients should auto bootstrap from env and these certs and the token

Reply:
projected volume cannot be used here, because temporary directory used by chroot env is not created at the time of pod creation, it's created in runtime. Overall I bind mounted required secrets from container into chroot env, as a result kubeconfig isn't needed anymore.